utils/bazel/llvm-project-overlay/llvm/config.bzl

```diff
@@ -2,118 +2,331 @@
+    cmakedefine_unset("HAVE_PTHREAD_GET_NAME_NP") |  # TODO: Likely wrong?
+    cmakedefine_unset("HAVE_PTHREAD_SET_NAME_NP") |  # TODO: Likely wrong?
+    cmakedefine("HAVE_MACH_MACH_H", enable = "@platforms//os:macos") |
+    cmakedefine_unset("HAVE_MALLCTL") |
+    cmakedefine("HAVE_MALLINFO", enable = "@platforms//os:linux") |
+    # TODO(aaronmondal): Make configurable and enable by default.
+    cmakedefine_unset("HAVE_MALLINFO2") |
+    cmakedefine("HAVE_MALLOC_MALLOC_H", enable = "@platforms//os:macos") |
+    cmakedefine(
+        "HAVE_MALLOC_ZONE_STATISTICS",
+        enable = "@platforms//os:macos",
+    ) |
+    cmakedefine_set("HAVE_POSIX_SPAWN") |
+    cmakedefine_set("HAVE_PREAD") |
+    cmakedefine("HAVE_PTHREAD_H", enable = "//llvm/config:posix") |
+    cmakedefine_set("HAVE_PTHREAD_MUTEX_LOCK") |
+    cmakedefine_set("HAVE_PTHREAD_RWLOCK_INIT") |
+    cmakedefine("HAVE_SBRK", enable = "@platforms//os:linux") |
+    cmakedefine("HAVE_SETENV", enable = "//llvm/config:posix") |
+    cmakedefine_set("HAVE_SIGALTSTACK") |
+    cmakedefine("HAVE_STRERROR_R", enable = "//llvm/config:posix") |
+    cmakedefine_set("HAVE_SYSCONF") |
+    cmakedefine_set("HAVE_SYS_MMAN_H") |
+    cmakedefine_unset("HAVE_STRUCT_STAT_ST_MTIMESPEC_TV_NSEC") |
+    cmakedefine(
+        "HAVE_STRUCT_STAT_ST_MTIM_TV_NSEC",
+        enable = "@platforms//os:linux",
+    ) |
+    cmakedefine("HAVE_UNISTD_H", enable = "//llvm/config:posix") |
+    cmakedefine_unset("HAVE_VALGRIND_VALGRIND_H") |
+    cmakedefine_unset("HAVE__ALLOCA") |
+    cmakedefine_unset("HAVE__CHSIZE_S") |
+    cmakedefine_set("HAVE__UNWIND_BACKTRACE") |
+    cmakedefine_unset("HAVE___ALLOCA") |
+    cmakedefine_unset("HAVE___ASHLDI3") |
+    cmakedefine_unset("HAVE___ASHRDI3") |
+    cmakedefine_unset("HAVE___CHKSTK") |
+    cmakedefine_unset("HAVE___CHKSTK_MS") |
+    cmakedefine_unset("HAVE___CMPDI2") |
+    cmakedefine_unset("HAVE___DIVDI3") |
+    cmakedefine_unset("HAVE___FIXDFDI") |
+    cmakedefine_unset("HAVE___FIXSFDI") |
+    cmakedefine_unset("HAVE___FLOATDIDF") |
+    cmakedefine_unset("HAVE___LSHRDI3") |
+    cmakedefine_unset("HAVE___MAIN") |
+    cmakedefine_unset("HAVE___MODDI3") |
+    cmakedefine_unset("HAVE___UDIVDI3") |
+    cmakedefine_unset("HAVE___UMODDI3") |
+    cmakedefine_unset("HAVE____CHKSTK") |
+    cmakedefine_unset("HAVE____CHKSTK_MS") |
+    {
+        '#cmakedefine HOST_LINK_VERSION "${HOST_LINK_VERSION}"': "/* #undef HOST_LINK_VERSION */",
+        '#cmakedefine LLVM_TARGET_TRIPLE_ENV "${LLVM_TARGET_TRIPLE_ENV}"': "/* #undef LLVM_TARGET_TRIPLE_ENV */",
+    } |
+    cmakedefine01_on("LLVM_VERSION_PRINTER_SHOW_HOST_TARGET_INFO") |
+    cmakedefine01_on("LLVM_VERSION_PRINTER_SHOW_BUILD_CONFIG") |
+    cmakedefine_unset("LLVM_ENABLE_LIBXML2") |
+    select({
+        "@platforms//os:windows": cmakedefine_string("LTDL_SHLIB_EXT", ".dll"),
+        "@platforms//os:macos": cmakedefine_string("LTDL_SHLIB_EXT", ".dylib"),
```

Comment:
FYI @keith Previously this and `LLVM_PLUGIN_EXT` were set to `.so` on macos which didn't match CMake.